hibernate-core/src/main/java/org/hibernate/tool/schema/internal/DdlTransactionIsolatorProvidedConnectionImpl.java

```diff
@@ -49,5 +55,17 @@ public void prepare() {
 
 	@Override
 	public void release() {
+		JdbcConnectionAccess connectionAccess = jdbcContext.getJdbcConnectionAccess();
```

Comment:
I see that `connectionAccessProvidedConnection` isn't used. Is the check on `connectionAccess` and the cast there just to make sure that the connection was provided?

Reply:
The cast is not needed and should be removed. However, we need to check to make sure the connection access was provided (e.g. `JdbcConnectionAccessProvidedConnectionImpl`) and only then release the connection.